File: solutions/1017. Convert to Base -2/1017.java
class Solution {
  public String baseNeg2(int n) {
    StringBuilder sb = new StringBuilder();

    while (n != 0) {
      sb.append(n % 2);
      n = -(n >> 1);
    }

    return sb.length() > 0 ? sb.reverse().toString() : "0";
  }
}

File: solutions/1017. Convert to Base -2/1017.cpp
class Solution {
 public:
  string baseNeg2(int n) {
    string ans;

    while (n != 0) {
      ans += to_string(n & 1);
      n = -(n >> 1);
    }

    return ans.empty() ? "0" : string{ans.rbegin(), ans.rend()};
  }
};
